futures-executor/src/local_pool.rs

```diff
@@ -2,21 +2,21 @@ use std::prelude::v1::*;
 
 use std::cell::{RefCell};
 use std::rc::{Rc, Weak};
+use std::mem::PinMut;
+use std::marker::Unpin;
 
-use futures_core::{Future, Poll, Async, Stream};
-use futures_core::task::{Context, Waker, LocalMap};
-use futures_core::executor::{Executor, SpawnError};
-use futures_core::never::Never;
+use futures_core::{Future, CoreFutureExt, Poll, Stream};
+use futures_core::task::{Context, LocalWaker, TaskObj, local_waker_from_nonlocal};
+use futures_core::executor::{Executor, SpawnObjError, SpawnErrorKind};
 use futures_util::stream::FuturesUnordered;
 use futures_util::stream::StreamExt;
 
 use thread::ThreadNotify;
 use enter;
 use ThreadPool;
 
-struct Task {
-    fut: Box<Future<Item = (), Error = Never>>,
-    map: LocalMap,
+struct TaskContainer {
```

Comment:
Can you add a comment to this that describes what it's for? Something like "wrapper for TaskObj that implements `Future` with a unit output" would help-- it's less obvious what its purpose is now that it no longer pairs the task with a LocalMap.

N.B. I think that TaskObj should implement `Future`-- we should add that impl in std. Opened https://github.com/rust-lang/rust/issues/51667

Reply:
Already made TaskObj implement Future

I‘ll add the comment

Reply:
I'm getting rid of TaskContainer entirely. We can use `TaskObj` directly. Can't do that in `ThreadPool`, but here it's simple